edgedb-client/src/builder.rs

```diff
@@ -1016,10 +1023,18 @@ impl Builder {
                 let conn = TcpStream::connect(
                     &(&self.host[..], self.port)
                 ).await.map_err(ClientConnectionError::with_source)?;
-                let host = if IpAddr::from_str(&self.host).is_ok() {
+                let host = if DNSNameRef::try_from_ascii_str(
```

Comment:
Please make a variable. Generally multi-ine `if`'s should be avoided, especially in the expression context.

And `x(\n  y\n)` formatting should be avoided at all costs in as part of the header of `if`, `while`, `match` etc, statement. As it makes a lot of distance between the opening keyword and an opening block of code *and* also makes idents ugly. Making code hard to read.